src/core/lv2/Lv2Proc.cpp

```diff
@@ -74,8 +74,11 @@ Plugin::PluginTypes Lv2Proc::check(const LilvPlugin *plugin,
 	AutoLilvNodes reqFeats(lilv_plugin_get_required_features(plugin));
 	LILV_FOREACH (nodes, itr, reqFeats.get())
 	{
-		issues.emplace_back(featureNotSupported,
-			lilv_node_as_string(lilv_nodes_get(reqFeats.get(), itr)));
+		const char* featName = lilv_node_as_string(
+								lilv_nodes_get(reqFeats.get(), itr));
+		if(		strcmp(featName, LV2_URID__map)
+			&&	strcmp(featName, LV2_URID__unmap))
```

Comment:
There's already a data structure containing the URIDs of all the supported features: the member `Lv2Proc::m_features`. Unfortunatelly, this code snippet is from the `static Lv2Proc::check`, where `m_features` is not available. I asked in the Lv2 IRC channel whether it makes sense to pass the same list of features to all plugins (whe using `lilv_plugin_instantiate`), and they said yes, so I think that `m_features` can be moved into `Lv2Manager`, and then in this static code snippet, we should be able to access it through `Engine::getLv2Manager`. Then, we can implement your suggested for-loop for a check here.

Does this make sense to you?

This should still work with your suggestions (below) to use heap-allocated `Lv2Feature` objects.

Reply:
OK, this was slightly wrong. You can pass the same *feature URIs* to multiple plugins, but not the same *feature data* (`LV2_Feature::data`) - at least not for all kind of features (for URID, it works).

Reply:
Since plugin initialization is not runtime critical (unlike maybe `Lv2Proc::check`, which is executed for 100s of plugins), I'd do it like this:

1. `Lv2Manager` contains (and returns a const reference to) an `std::set<const char *>` listing all supported feature URIs.
2. `Lv2Proc::check` uses this `set` directly to lookup if a required plugin is not supported.
3. `Lv2Proc::initPlugin`
    1. fetches this set
    2. copies it into a (function-local) `std::map` using a loop
    3. fills the map like `map[LV2_URID__map] = man->uridMap().mapFeature()`
    4. fills a vector `m_features` by copying in a second loop (`std::pair` -> `LV2_Feature`)
    5. finally fills a vector `m_featurePointers` with the pointers from `m_features`
    6. passes `m_featurePointers.data()` to lilv_plugin_instantiate

Do you think this will be good?

Reply:
I think this mostly makes sense and sounds good. A couple of questions/points:
* `std::set<const char *>` will compare based on pointer value - you'll need to provide a custom comparator, or use a `std::set<std::string>`.
* What's the advantage to fetching the URI set (3.i) and copying it into a map (3.ii), if the map then needs to be manually populated anyway (3.iii)?
* Could all the features stuff perhaps be factored out into an `Lv2Features` class or something? E.g. call `Lv2Features::isFeatureSupported` for each required feature in `Lv2Proc::check`, and in `Lv2Proc` have an `Lv2Features m_features;` and pass `m_features.features()` to `lilv_plugin_instantiate`. Or would this be awkward when trying to initialise instance-specific feature data?

Reply:
Note to self: Addressed first and second points in 2 new commits.

Reply:
Third point also fixed. It should not be awkward, see use of `Lv2Features::operator[]`

Reply:
All fixed in 1cfa713. Resolving.